src/frontend/src/flows/addDevice/manage/addDevicePickAlias.ts

```diff
@@ -51,4 +53,26 @@ const init = (): Promise<string | null> =>
       event.preventDefault();
       resolve(deviceAlias.value);
     };
+    const deviceInput = document.getElementById(
+      "deviceAlias"
+    ) as HTMLInputElement;
+
+    deviceInput.addEventListener("invalid", () => {
+      const firstChar = deviceInput.value[0];
+      const lastChar = deviceInput.value.slice(-1);
+
+      if (firstChar === " ") {
```

Comment:
the `pattern` above (`^[A-Za-z0-9_]+((-|\\s)*[A-Za-z0-9_])*$`) already seems to preclude starting with a space; which validity check takes precedence here? Same question below with the checks below.

Reply:
@nmattia I've made the changes, with validation order of precedence

1. If no entries (validity.valueMissing)

Following all fall under validity.patternMismatch

2. If name starts with space
3. If it ends with space or hyphen 
4.  If there are special characters


https://user-images.githubusercontent.com/38238542/185503386-c73e5839-fd9f-40de-aad4-e0519716d189.mov